Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -171,7 +172,7 @@ public void reorderNodesInBreadthFirstOrder() {
                     if (isLeaf(leftChild)) {
                         // the parent is the current node and the indices, mass are being copied over
                         map[leftChild] = result.addLeaf(currentNode, getPointReference(leftChild), getMass(leftChild));
-                        validateInternalState(map[leftChild] == leafcounter, "incorrect state");
+                        // validateInternalState(map[leftChild] == leafcounter, "incorrect state");
```

Comment:
Should we delete these instead of commenting them out?

Reply:
The best solution would be a debug-flag. If we keep them uncommented, then these assertions are untestable (because they're holding!) and branch/code coverage is low. If we delete them then we lose the insight. We can delete them if we choose to.

Reply:
In a future PR, we can replace checks like these with asserts, which must be explicitly enabled at runtime by passing a command-line flag to the JVM.